renderer.go

```diff
@@ -0,0 +1,417 @@
+package lipgloss
+
+import (
+	"fmt"
+	"strings"
+	"unicode"
+
+	"github.com/muesli/reflow/truncate"
+	"github.com/muesli/reflow/wordwrap"
+	"github.com/muesli/reflow/wrap"
+	"github.com/muesli/termenv"
+)
+
+var renderer = NewRenderer()
+
+// Renderer is a lipgloss terminal renderer.
+type Renderer struct {
+	output            *termenv.Output
+	hasDarkBackground bool
+}
+
+// RendererOption is a function that can be used to configure a Renderer.
+type RendererOption func(r *Renderer)
+
+// DefaultRenderer returns the default renderer.
+func DefaultRenderer() *Renderer {
+	return renderer
+}
+
+// NewRenderer creates a new Renderer.
+func NewRenderer(options ...RendererOption) *Renderer {
+	r := &Renderer{}
+	for _, option := range options {
+		option(r)
+	}
+	if r.output == nil {
+		r.output = termenv.DefaultOutput()
+	}
+	if !r.hasDarkBackground {
+		r.hasDarkBackground = r.output.HasDarkBackground()
+	}
+	return r
+}
+
+// WithOutput sets the termenv Output to use for rendering.
+func WithOutput(output *termenv.Output) RendererOption {
+	return func(r *Renderer) {
+		r.output = output
+	}
+}
+
+// WithDarkBackground forces the renderer to use a dark background.
+func WithDarkBackground() RendererOption {
+	return func(r *Renderer) {
+		r.SetHasDarkBackground(true)
+	}
+}
+
+// ColorProfile returns the detected termenv color profile.
+func (r *Renderer) ColorProfile() termenv.Profile {
+	return r.output.Profile
+}
+
+// ColorProfile returns the detected termenv color profile.
+func ColorProfile() termenv.Profile {
+	return renderer.ColorProfile()
+}
+
+// SetColorProfile sets the color profile. This function exists mostly for
+// testing purposes so that you can assure you're testing against a specific
+// profile.
+//
+// Outside of testing you likely won't want to use this function as
+// ColorProfile() will detect and cache the terminal's color capabilities
+// and choose the best available profile.
+//
+// Available color profiles are:
+//
+// termenv.Ascii (no color, 1-bit)
+// termenv.ANSI (16 colors, 4-bit)
+// termenv.ANSI256 (256 colors, 8-bit)
+// termenv.TrueColor (16,777,216 colors, 24-bit)
+//
+// This function is thread-safe.
+func (r *Renderer) SetColorProfile(p termenv.Profile) {
```

Comment:
Does it make sense to have a `func WithColorProfile() RendererOption` in the same way we have `WithOutput` and `WithDarkBackground`?

Reply:
I like that.

Reply:
Cool, just added.